### PR TITLE
Change high resolution image with preview

### PR DESCRIPTION
**Pull request by wildan-m**
### Details

We have observed significant lag when displaying and interacting with high-resolution images in the modal. This PR aims to modify the behavior when handling high-resolution images.

### Fixed Issues

$ https://github.com/Expensify/App/issues/38180
PROPOSAL: https://github.com/Expensify/App/issues/38180#issuecomment-2126395141

#### Standard resolution image
1. Go to any report chat
2. Select + Icon and select `Add Attachment`
3. Upload this image ([4000x4000.png](https://github.com/Expensify/App/assets/11847279/64a9ae21-827a-4dd5-bb83-a1f17927aecf))
4. Verify original image displayed and zoomable
5. Select `Send` button
6. Select successfully uploaded image
7. Verify original image displayed and zoomable
8. If there are multiple attachments, verify it can be navigated through arrow buttons and swipe/drag action

- [x] Verify that no errors appear in the JS console

#### High resolution image
1. Go to any report chat
2. Select + Icon and select `Add Attachment`
3. Upload this image ([10000px](https://github.com/Expensify/App/assets/11847279/d2438f8c-3616-4c93-85e0-6dc2d47d618b))
4. Verify that image name shown with left image icon
5. Verify that there is info text at the bottom of the image with text: `This image is too large to preview before uploading.`
6. Select `Send` button
7. Select successfully uploaded image
8. Verify thumbnail image displayed (1024px version) and zoomable
9. Verify that there is info text at the bottom of the image with text: `This image has been resized for previewing. Download for full resolution.`
10. If there are multiple attachments, ensure they can be navigated using arrow buttons and swipe/drag actions, with the information text appearing or hiding based on the image resolution threshold (5000px)
- [x] Verify that no errors appear in the JS console

#### Standard resolution image
1. Turn off the network
2. Go to any report chat
3. Select + Icon and select `Add Attachment`
4. Upload this image ([4000x4000.png](https://github.com/Expensify/App/assets/11847279/64a9ae21-827a-4dd5-bb83-a1f17927aecf))
5. Verify original image displayed and zoomable
6. Select `Send` button
7. The `Uploading attachment..` text will appear
8. Turn on the network
9. Select successfully uploaded image
11. Verify original image displayed and zoomable
12. If there are multiple attachments, verify it can be navigated through arrow buttons and swipe/drag action

- [x] Verify that no errors appear in the JS console

#### High resolution image
1. Turn off the network
2. Go to any report chat
3. Select + Icon and select `Add Attachment`
4. Upload this image ([10000px](https://github.com/Expensify/App/assets/11847279/d2438f8c-3616-4c93-85e0-6dc2d47d618b))
5. Verify that only image name shown with left image icon
6. Select `Send` button
7. The `Uploading attachment..` text will appear
8. Turn on the network
9. Select successfully uploaded image
10. Verify thumbnail image displayed (1024px version) and zoomable
11. Verify that there is info text at the bottom of the image with text: `This image has been resized for previewing. Download for full resolution.`
13. If there are multiple attachments, ensure they can be navigated using arrow buttons and swipe/drag actions, with the information text appearing or hiding based on the image resolution threshold (5000px)
- [x] Verify that no errors appear in the JS console

### QA Steps

Same as Test
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message: https://expensify.slack.com/archives/C01GTK53T8Q/p1719953316322399
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/11847279/fc2a685a-7b51-4534-8018-01a66656cb29

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/11847279/6358f983-d874-4515-971a-1f084889e452

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/11847279/033375aa-6f08-45e3-8c88-949b5d61d472

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/11847279/11427370-6168-4d59-a485-1348747832d5

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/11847279/46ca50de-0bd6-4f27-9cdb-2c768de85a3d

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/11847279/86fbabb0-2d7c-4207-9e35-aaccfd8d11da

</details>

### CODE REVIEW

**Comment by wildan-m** [comment]
@Expensify/design

**Comment by dannymcclain** [comment]
I'm wondering if on this screen we should put some kind of message letting users know that the file is too big to preview before upload. @Expensify/design what do you think about something like that?

If you watch the video it feels weird that you get a preview for the 4000x4000 image and then don't for this one (with no explanation).  

CleanShot 2024-07-04 at 08 36 22@2x

**Comment by shawnborton** [comment]
That makes sense to me, perhaps we could use a similar style for the message as we're using once you view it post-upload?

**Comment by shawnborton** [comment]
Nice, I think that looks great!

**Comment by trjExpensify** [comment]
Nice improvement. #context!

**Comment by wildan-m** [comment]
@sobitneupane updated code with refactoring and added info text for non-uploaded high-resolution images ([ref](https://github.com/Expensify/App/pull/44807#issuecomment-2209062658))

File: src/CONST.ts [hunk 1]
```diff
@@ -570,6 +570,7 @@ const CONST = {
     GOOGLE_MEET_URL_ANDROID: 'https://meet.google.com',
     GOOGLE_DOC_IMAGE_LINK_MATCH: 'googleusercontent.com',
     IMAGE_BASE64_MATCH: 'base64',
+
     DEEPLINK_BASE_URL: 'new-expensify://',
```
Comment (Julesssss) [inline src/CONST.ts:574]: NAB: We prefer to not make unrelated changes on our PRs.

Comment (wildan-m) [reply src/CONST.ts]: reverted

File: src/components/Attachments/AttachmentCarousel/Pager/index.tsx [hunk 2]
```diff
@@ -70,13 +73,27 @@ function AttachmentCarouselPager(
...
+        () => isPagerScrolling.value,
+        (isScrolling, previousIsScrolling) => {
+            if (isScrolling === previousIsScrolling) {
+                return;
+            }
+            runOnJS(onIsPagerScrollingChange ?? (() => {}))(isScrolling);
+        },
+        [onIsPagerScrollingChange],
+    );
+
```
Comment (sobitneupane) [inline src/components/Attachments/AttachmentCarousel/Pager/index.tsx:86]: @wildan-m Why is this change required?

File: src/CONST.ts [hunk 3]
```diff
@@ -1201,6 +1201,8 @@ const CONST = {
         NOTE: 'n',
     },
 
+    IMAGE_HIGH_RESOLUTION_THRESHOLD: 5000,
```
Comment (sobitneupane) [inline src/CONST.ts:1204]: How did we end of choosing 5000 as the threshold?

File: src/components/AttachmentModal.tsx [hunk 4]
```diff
@@ -183,7 +187,7 @@ function AttachmentModal({
     const nope = useSharedValue(false);
     const isOverlayModalVisible = (isReceiptAttachment && isDeleteReceiptConfirmModalVisible) || (!isReceiptAttachment && isAttachmentInvalid);
     const iouType = useMemo(() => (isTrackExpenseAction ? CONST.IOU.TYPE.TRACK : CONST.IOU.TYPE.SUBMIT), [isTrackExpenseAction]);
-
+    const [isAttachmentCarouselScrolling, setIsAttachmentCarouselScrolling] = useState(false);
```
Comment (sobitneupane) [inline src/components/AttachmentModal.tsx:190]: Can you please explain what is this state for and why is it required?

File: src/CONST.ts (hunk 3 above)
Comment (wildan-m) [reply src/CONST.ts:1204]: I use [this suggestion](https://github.com/Expensify/App/issues/38180#issuecomment-2191136178) as reference, should we increase it?

cc: @Julesssss

**Review by sobitneupane — COMMENTED** [review]
Arrow moves up for the attachment with size info.


https://github.com/Expensify/App/assets/25876548/85732848-2d4e-49f2-a4fc-0237c18739c3

File: src/components/Attachments/AttachmentCarousel/Pager/index.tsx (hunk 2 above)
Comment (wildan-m) [reply src/components/Attachments/AttachmentCarousel/Pager/index.tsx:86]: We need to adjust the scrolling state so it can be utilized in AttachmentModal

On iOS, tapping the carousel arrow currently causes a crash as it simultaneously toggles the highRes caption

The explanation is here:

https://github.com/Expensify/App/blob/4bf56529ab98f2c7725a09baeba3f8bb668f9095/src/components/AttachmentModal.tsx#L204

File: src/components/AttachmentModal.tsx (hunk 4 above)
Comment (wildan-m) [reply src/components/AttachmentModal.tsx:190]: The explanation is the same as [here](https://github.com/Expensify/App/pull/44807#discussion_r1669923321).

File: src/CONST.ts (hunk 3 above)
Comment (Julesssss) [reply src/CONST.ts:1204]: Ah, to be honest that was a bit of a guess. I'm open to suggestions here

Comment (Julesssss) [reply src/CONST.ts:1204]: @wildan-m lets start with 7000px and match Slack.

**Comment by wildan-m** [comment]
> Arrow moves up for the attachment with size info.

That's because it's not part of carousel content; moving the carousel container along with the arrow icon. This issue can be avoided if HighResInfo is included in the content, but there is a drawback: the text will move each time we swipe/navigate.

https://github.com/Expensify/App/assets/11847279/061cd5a6-c725-4698-bffd-612563ea6e8a

@sobitneupane @Julesssss @dannymcclain Should we include the text in the carousel item content? If it is part of the item content, we do not need to consider the scrolling state [here](https://github.com/Expensify/App/pull/44807#discussion_r1669923321).

**Comment by Julesssss** [comment]
> This issue can be avoided if HighResInfo is included in the content, but there is a drawback: the text will move each time we swipe/navigate.

@wildan-m just to clarify, the problem you are highlighting is that the 'This image has been resized...' text is locked to the image, and does not remain in place? Unless I misunderstand, this looks great to me in the video and I would expect this to happen, rather than have the text remain static and dissapear/reappear depending on the content. 

@dannymcclain it would be good to get your thoughts here though.

**Comment by wildan-m** [comment]
> just to clarify, the problem you are highlighting is that the 'This image has been resized...' text is locked to the image, and does not remain in place?

@Julesssss that is correct, I would also prefer that method as it will not cause any delays in displaying/hiding the text, especially for native.

**Comment by dannymcclain** [comment]
>Unless I misunderstand, this looks great to me in the video and I would expect this to happen, rather than have the text remain static and dissapear/reappear depending on the content.

Totally agree—I feel like this is how I would expect it to work!

**Comment by wildan-m** [comment]
@sobitneupane The PR updated.

- Move the HighResInfo into attachmentview
- Remove changes related to attachment carousel listener ([ref](https://github.com/Expensify/App/pull/44807#discussion_r1669923321))
- Resolve issue arrow moves ([ref](https://github.com/Expensify/App/pull/44807#pullrequestreview-2165347854))

**Comment by sobitneupane** [comment]
@wildan-m The app crashes when I try to open an attachment. There are few console errors as well. Can you please test it in your end?

**Comment by wildan-m** [comment]
@sobitneupane I have tested them, have you made sure to set `USE_REACT_STRICT_MODE` to false?

It's in src/CONFIG.ts

Some people have also experienced the same thing:
https://expensify.slack.com/archives/C01GTK53T8Q/p1719513440587919

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/25876548/89d10699-900e-4e76-9e9c-058867dab68c

https://github.com/Expensify/App/assets/25876548/15caf1b5-99f6-432c-a354-246ac06aeaa9

</details>


<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/25876548/96bb3c70-3fbd-405b-aca7-481e1fc8f28b


</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/25876548/bba78686-6b05-41f6-9e2f-0eb9ab427367


https://github.com/Expensify/App/assets/25876548/ceacddf7-5f7c-4472-a891-b4eb2763f6a2


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/25876548/1ef825b8-b619-4c8c-8c84-b0de02412acc

https://github.com/Expensify/App/assets/25876548/7932d60a-1734-44e7-ac6c-93170bc3a138



</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/25876548/f1add70b-8de8-4fe3-a490-86537bd2e022


</details>

**Comment by dubielzyk-expensify** [comment]
Maybe I've missed something, but do we really need this warning?

**Comment by dubielzyk-expensify** [comment]
Maybe I've missed something, but do we really need this warning?

CleanShot 2024-07-10 at 20 25 34@2x

I agree that we should add the warning Danny suggested above [here](https://github.com/Expensify/App/pull/44807#issuecomment-2209062658), but I don't think the user cares that we're downscaling their preview. Feels a bit too technical for the average user. 

Keen to hear if @Expensify/design agrees.

**Comment by sobitneupane** [comment]
@wildan-m App is quite slow on IOS and it keeps crashing on Android. It might not be related to the PR. But Can you please retest it in your end?

Also, lets add some padding in the warning message (especially on Android/native and IOS/native)

Screenshot 2024-07-10 at 16 01 07 Screenshot 2024-07-10 at 16 23 17

**Comment by Julesssss** [comment]
> @wildan-m App is quite slow on IOS and it keeps crashing on Android. It might not be related to the PR. But Can you please retest it in your end?

@sobitneupane just to confirm, the images you're using are all bigger than 7000*7000?

**Comment by shawnborton** [comment]
Hmm @dubielzyk-expensify I do think you make a totally fair point... I don't feel too strongly one way or the other.

**Comment by dannymcclain** [comment]
>I don't think the user cares that we're downscaling their preview. Feels a bit too technical for the average user.

I think the reason we decided to include this is because the zooming/panning is going to look much much worse on the resized image vs. the high res image the user uploaded. But if you feel like really no one will care we can ditch it. Alternatively we could change it to just say `This image has been resized. Download for full resolution.`

**Comment by dubielzyk-expensify** [comment]
> I think the reason we decided to include this is because the zooming/panning is going to look much much worse on the resized image vs. the high res image the user uploaded. But if you feel like really no one will care we can ditch it. Alternatively we could change it to just say This image has been resized. Download for full resolution.

It's not like we're showing the a 300x300 image though? Isn't just the preview more manageable but still reasonably decent? 

I guess I kinda just expect all internet services to process image or resize them. Feels like it'd be weird if Instagram or Slack said something similar which is why I'm calling it out. I guess this warning only shows for very high resolution so maybe it's fine and more transparent. 

I trust you know this better than me. Just wanted to add my perspective 😄

**Comment by wildan-m** [comment]
@dubielzyk-expensify Consider [this case](https://github.com/Expensify/App/issues/38180#issuecomment-2158157452) where there are significant differences between width and height. Since the preview image version should maintain its aspect ratio, the compressed image might not be readable even when zoomed in. Without text describing that a higher resolution is available for download, the user might assume that the downloaded version will also be blurry.

https://github.com/Expensify/App/assets/11847279/756ae0c1-3e7c-4fa3-b7d5-89b4cdcdd808

**Comment by dubielzyk-expensify** [comment]
Very fair point! Hadn't considered that. 

Sorry if I missed this earlier, what's the threshold for when we show this preview warning then? Is it based on the width/height of the image? If so what's the threadshold?

**Comment by wildan-m** [comment]
@dubielzyk-expensify currently 7000 pixels in either width or height. ([ref](https://github.com/Expensify/App/pull/44807#discussion_r1670351125))

**Comment by wildan-m** [comment]
@sobitneupane The [PR](https://github.com/Expensify/App/pull/44807#issue-2390055344) has been updated. 

- Update video test
- Adjust padding for native
- Merge main

**Comment by dannymcclain** [comment]
>I guess I kinda just expect all internet services to process image or resize them. Feels like it'd be weird if Instagram or Slack said something similar which is why I'm calling it out.

Totally see what you mean, but I think what we have here still makes sense. Slack for example just doesn't show you a preview AT ALL for really big images. 

CleanShot 2024-07-11 at 08 25 54@2x

CleanShot 2024-07-11 at 08 26 03@2x

If you want to see the image, you have to download it. We're kinda taking a different approach where we still want you to be able to see _something_, but I think it's important to let you know that it is not the actual/proper image file, so it's going to be degraded a bit.

>I trust you know this better than me. Just wanted to add my perspective 😄

LOL I'm not sure I'd go as far as to say I know better 😂 but I do appreciate you sharing your perspective! Again, I think this treatment still makes sense because it takes a massively large image to even cause this behavior. So I'd imagine most people will never even see this at all.

**Comment by dubielzyk-expensify** [comment]
Makes sense. Thanks for going on the journey. Having seen [this example](https://github.com/Expensify/App/pull/44807#issuecomment-2222015009) I see the value in the UI given so many of our attachments are likely to be long or PDF like.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/user-attachments/assets/4c17fae7-09d0-482e-aee7-b574187ec0c4


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/user-attachments/assets/718d78ca-64f1-43fa-8a48-52bc8cb8095e


https://github.com/user-attachments/assets/86257efe-b8ba-4eb3-9861-a37099abcc5a


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/3e945ead-99a9-4e30-aa87-1c623a40cc51


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/11e1a508-9aaa-45fc-ad12-ba40bd4571c2


https://github.com/user-attachments/assets/7db98e41-16c6-4bfa-aefe-edbd2e8d9f60




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/2d61df6a-fa84-4d34-8c2b-395d23cedbdb

</details>

**Comment by shawnborton** [comment]
Screenshots feel pretty good to me, let's see what the rest of the @Expensify/design team thinks too.

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by Julesssss — APPROVED** [review]
This is looking good. Lets give @dannymcclain and @dubielzyk-expensify a final chance to review (the test builds can be found above)

**Comment by dannymcclain** [comment]
Looking good to me!

**Review by dubielzyk-expensify — APPROVED** [review]
LGTM

**Review by Julesssss — APPROVED** [review]

**Comment by Julesssss** [comment]
Thanks all for your help!
